### ISSUE TITLE
[build]: Critical dependency Error When building with Webpack

### ISSUE THREAD

**Issue by sfsf-xweb-sh** [SAP/ui5-webcomponents#8410]
### Bug Description

hi, team, 

We're facing the Critical dependency Error When building with Webpack

## Background:  
What's is the `Critical dependency` Issue? [link](https://guidingcode.com/webpack-critical-dependency-the-request-of-a-dependency-is-an-expression/)

## Current Behavior:  
It will compile failed with below messages

```sh
[2024-03-08T07:12:06.740Z] Treating warnings as errors because process.env.CI = true.
[2024-03-08T07:12:06.740Z] Most CI servers set it automatically.
[2024-03-08T07:12:06.740Z] Failed to compile.
[2024-03-08T07:12:06.740Z] ./node_modules/@ui5/webcomponents-localization/dist/sap/base/util/resolveReference.js
[2024-03-08T07:12:06.740Z] Critical dependency: the request of a dependency is an expression
[2024-03-08T07:12:06.740Z]  script returned exit code 1
```
> It's warning in local and error in CI environment.

And here is the link explain by the webpack https://github.com/webpack/webpack/issues/10496

### Steps to Reproduce

1.
2.
3.
...

### Priority

None

### UI5 Web Components Version

1.22.0

### Browser

Chrome, Edge, Firefox, Safari

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by IlianaB** [comment]
Hello @SAP/ui5-webcomponents-topic-core , please have a look at this issue.

Regards,
Iliana

**Comment by pskelin** [comment]
a fix was provided with #8359, which is part of [1.23.0](https://github.com/SAP/ui5-webcomponents/releases/tag/v1.23.0)
